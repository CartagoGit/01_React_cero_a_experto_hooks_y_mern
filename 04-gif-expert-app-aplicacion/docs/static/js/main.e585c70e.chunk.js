(this["webpackJsonp04-gif-expert-app-aplicacion"]=this["webpackJsonp04-gif-expert-app-aplicacion"]||[]).push([[0],{14:function(t,e,n){},15:function(t,e,n){"use strict";n.r(e);var c=n(1),r=n(5),a=n.n(r),i=n(2),s=n(0),u=function(t){var e=t.categories,n=t.setCategories,r=Object(c.useState)(e),a=Object(i.a)(r,2),u=a[0],o=a[1];return Object(s.jsx)("form",{onSubmit:function(t){t.preventDefault(),u.trim().length>0?(n((function(t){return[u]})),o("")):o("")},children:Object(s.jsx)("input",{type:"text",value:u,onChange:function(t){o(t.target.value)}})})},o=n(7),j=n(4),d=n.n(j),p=n(6),l=function(){var t=Object(p.a)(d.a.mark((function t(e){var n,c,r,a,i;return d.a.wrap((function(t){for(;;)switch(t.prev=t.next){case 0:return n="".concat("https://api.giphy.com/v1/gifs/search","?limit=").concat(10,"&q=").concat(encodeURI(e),"&api_key=").concat("z56j7qt1NchrW75KsQCI2ysX5sfLWZtJ"),t.next=3,fetch(n);case 3:return c=t.sent,t.next=6,c.json();case 6:return r=t.sent,a=r.data,i=a.map((function(t){return{id:t.id,title:t.title,url:t.images.downsized_medium.url}})),t.abrupt("return",i);case 10:case"end":return t.stop()}}),t)})));return function(e){return t.apply(this,arguments)}}(),f=function(t){t.id;var e=t.title,n=t.url;return Object(s.jsxs)("div",{className:"card animate__tada",children:[Object(s.jsx)("img",{src:n,alt:e}),Object(s.jsx)("p",{children:e})]})},b=function(t){var e=t.category,n=function(t){var e=Object(c.useState)({data:[],loading:!0}),n=Object(i.a)(e,2),r=n[0],a=n[1];return Object(c.useEffect)((function(){l(t).then((function(t){a({data:t,loading:!1})}))}),[t]),r}(e),r=n.data,a=n.loading;return Object(s.jsxs)(s.Fragment,{children:[Object(s.jsx)("h3",{children:e}),a&&Object(s.jsx)("p",{children:"Cargando..."}),Object(s.jsx)("div",{className:"cardGrid",children:r.map((function(t){return Object(s.jsx)(f,Object(o.a)({},t),t.id)}))})]})},h=function(){var t=Object(c.useState)(["Busca un gif"]),e=Object(i.a)(t,2),n=e[0],r=e[1];return Object(s.jsxs)(s.Fragment,{children:[Object(s.jsx)("h2",{children:"GifExpertApp"}),Object(s.jsx)(u,{categories:n,setCategories:r}),Object(s.jsx)("hr",{}),Object(s.jsx)("ol",{children:n.map((function(t){return Object(s.jsx)(b,{category:t},t)}))})]})};n(14);a.a.render(Object(s.jsx)(h,{}),document.getElementById("root"))}},[[15,1,2]]]);
//# sourceMappingURL=main.e585c70e.chunk.js.map