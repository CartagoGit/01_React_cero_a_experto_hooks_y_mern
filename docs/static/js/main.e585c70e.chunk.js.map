{"version":3,"sources":["components/AddCategory.js","helpers/getGifs.js","components/GifGridItem.js","components/GifGrid.js","hooks/useFetchGifs.js","GifExpertApp.js","index.js"],"names":["AddCategory","categories","setCategories","useState","inputValue","setInputValue","onSubmit","e","preventDefault","trim","length","c","type","value","onChange","target","getGifs","category","a","url","encodeURI","fetch","resp","json","data","gifs","map","img","id","title","images","downsized_medium","GifGridItem","className","src","alt","GifGrid","loading","state","setState","useEffect","then","imgs","useFetchGif","GifExpertApp","ReactDOM","render","document","getElementById"],"mappings":"wNAGaA,EAAc,SAAC,GAAmC,IAAjCC,EAAgC,EAAhCA,WAAYC,EAAoB,EAApBA,cAAoB,EACzBC,mBAASF,GADgB,mBACtDG,EADsD,KAC1CC,EAD0C,KAiB7D,OACC,sBAAMC,SAXc,SAACC,GACrBA,EAAEC,iBAEEJ,EAAWK,OAAOC,OAAS,GAE9BR,GAAc,SAACS,GAAD,MAAO,CAACP,MACtBC,EAAc,KACRA,EAAc,KAIrB,SACC,uBAAOO,KAAK,OAAOC,MAAOT,EAAYU,SAhBd,SAACP,GAC1BF,EAAcE,EAAEQ,OAAOF,a,8BCHZG,EAAO,uCAAG,WAAOC,GAAP,uBAAAC,EAAA,6DACbC,EADa,UAJR,uCAIQ,kBAFT,GAES,cACuBC,UACtCH,GAFe,oBAHR,oCAGQ,SAKAI,MAAMF,GALN,cAKbG,EALa,gBAMIA,EAAKC,OANT,uBAMXC,EANW,EAMXA,KACFC,EAAOD,EAAKE,KAAI,SAACC,GACnB,MAAO,CACHC,GAAID,EAAIC,GACRC,MAAOF,EAAIE,MACXV,IAAKQ,EAAIG,OAAOC,iBAAiBZ,QAXtB,kBAiBZM,GAjBY,4CAAH,sDCFPO,EAAc,SAAC,GAAqB,EAApBJ,GAAqB,IAAjBC,EAAgB,EAAhBA,MAAOV,EAAS,EAATA,IAEpC,OACI,sBAAKc,UAAU,qBAAf,UAEI,qBAAKC,IAAKf,EAAKgB,IAAKN,IACpB,4BAAIA,QCFHO,EAAU,SAAC,GAAkB,IAAhBnB,EAAe,EAAfA,SAAe,ECHd,SAACA,GAAc,IAAD,EACdd,mBAAS,CAClCqB,KAAM,GACNa,SAAS,IAH8B,mBACjCC,EADiC,KAC1BC,EAD0B,KAexC,OATAC,qBAAU,WACTxB,EAAQC,GAAUwB,MAAK,SAACC,GACvBH,EAAS,CACRf,KAAMkB,EACNL,SAAS,SAGT,CAACpB,IAEGqB,EDH2BK,CAAY1B,GAAhCa,EAT0B,EAShCN,KAAca,EATkB,EASlBA,QAEtB,OACC,qCACC,6BAAKpB,IACKoB,GAAW,4CAIrB,qBAAKJ,UAAU,WAAf,SAIEH,EAAOJ,KAAI,SAACC,GAAD,OAGX,aAFA,CAEC,EAAD,eAA8BA,GAAZA,EAAIC,aE3BdgB,EAAe,WAAM,MAEGzC,mBAAS,CAAC,iBAFb,mBAE1BF,EAF0B,KAEdC,EAFc,KAUjC,OACC,qCACC,8CACA,cAAC,EAAD,CAAaD,WAAYA,EAAYC,cAAeA,IACpD,uBAEA,6BACED,EAAWyB,KAAI,SAACT,GAAD,OACf,cAAC,EAAD,CAAwBA,SAAUA,GAApBA,Y,MChBnB4B,IAASC,OAAO,cAAC,EAAD,IAAkBC,SAASC,eAAe,W","file":"static/js/main.e585c70e.chunk.js","sourcesContent":["import React, { useState } from \"react\";\r\nimport PropTypes from \"prop-types\";\r\n\r\nexport const AddCategory = ({ categories, setCategories }) => {\r\n\tconst [inputValue, setInputValue] = useState(categories);\r\n\t// console.log(categories);\r\n\tconst handleInputChange = (e) => {\r\n\t\tsetInputValue(e.target.value);\r\n\t};\r\n\r\n\tconst handleSubmit = (e) => {\r\n\t\te.preventDefault();\r\n\r\n\t\tif (inputValue.trim().length > 0) {\r\n\t\t\t// setCategories((c) => [inputValue, ...c]);\r\n\t\t\tsetCategories((c) => [inputValue]);\r\n\t\t\tsetInputValue(\"\");\r\n\t\t} else setInputValue(\"\");\r\n\t};\r\n\r\n\treturn (\r\n\t\t<form onSubmit={handleSubmit}>\r\n\t\t\t<input type='text' value={inputValue} onChange={handleInputChange} />\r\n\t\t</form>\r\n\t);\r\n};\r\n\r\nAddCategory.propTypes = {\r\n\tsetCategories: PropTypes.func.isRequired\r\n};\r\n","const apiWeb = \"https://api.giphy.com/v1/gifs/search\";\r\nconst apiKey = \"z56j7qt1NchrW75KsQCI2ysX5sfLWZtJ\";\r\nconst limit = 10;\r\n\r\nexport const getGifs = async (category) => {\r\n    const url = `${apiWeb}?limit=${limit}&q=${encodeURI(\r\n        category\r\n    )}&api_key=${apiKey}`;\r\n\r\n    const resp = await fetch(url);\r\n    const { data } = await resp.json();\r\n    const gifs = data.map((img) => {\r\n        return {\r\n            id: img.id,\r\n            title: img.title,\r\n            url: img.images.downsized_medium.url\r\n        };\r\n    });\r\n\r\n    // console.log(gifs);\r\n    // setImages(gifs);\r\n    return gifs;\r\n    // console.log(images);\r\n};","import React from 'react'\r\n\r\nexport const GifGridItem = ({id, title, url}) => {\r\n    // console.log(id,title,url);\r\n    return (\r\n        <div className='card animate__tada'>\r\n            {/* {props} */}\r\n            <img src={url} alt={title}/>\r\n            <p>{title}</p>\r\n        </div>\r\n    )\r\n}\r\n","// import React, { useState, useEffect } from \"react\";\r\nimport React from \"react\";\r\nimport { useFetchGif } from \"../hooks/useFetchGifs\";\r\n// import { getGifs } from \"../helpers/getGifs\";\r\nimport { GifGridItem } from \"./GifGridItem\";\r\n\r\nexport const GifGrid = ({ category }) => {\r\n\t// const [count, setCount] = useState(0);\r\n\t// const [images, setImages] = useState([]);\r\n\r\n\t// useEffect(() => {\r\n\t// \tgetGifs(category).then((imgs) => setImages(imgs));\r\n\t// \t//// eslint-disable-next-line react-hooks/exhaustive-deps\r\n\t// }, [category]);\r\n\r\n\tconst { data: images, loading } = useFetchGif(category);\r\n\r\n\treturn (\r\n\t\t<>\r\n\t\t\t<h3>{category}</h3>\r\n            {loading && <p>Cargando...</p>}\r\n\r\n\t\t\t\r\n\r\n\t\t\t<div className='cardGrid'>\r\n\t\t\t\t{/* { <h3>{count}</h3> } */}\r\n\t\t\t\t{/* <button onClick={ () => setCount(count+1)}></button> */}\r\n\r\n\t\t\t\t{images.map((img) => (\r\n\t\t\t\t\t// <li key={id}> {title}</li>\r\n\t\t\t\t\t// <GifGridItem key={img.id} img ={img}/>\r\n\t\t\t\t\t<GifGridItem key={img.id} {...img} />\r\n\t\t\t\t))}\r\n\t\t\t</div>\r\n\t\t</>\r\n\t);\r\n};\r\n","import { useEffect, useState } from \"react\";\r\nimport { getGifs } from \"../helpers/getGifs\";\r\n\r\nexport const useFetchGif = (category) => {\r\n\tconst [state, setState] = useState({\r\n\t\tdata: [],\r\n\t\tloading: true\r\n\t});\r\n\r\n\tuseEffect(() => {\r\n\t\tgetGifs(category).then((imgs) => {\r\n\t\t\tsetState({\r\n\t\t\t\tdata: imgs,\r\n\t\t\t\tloading: false\r\n\t\t\t});\r\n\t\t});\r\n\t}, [category]);\r\n\r\n\treturn state;\r\n};\r\n","import React, { useState } from \"react\";\r\nimport { AddCategory } from \"./components/AddCategory\";\r\nimport { GifGrid } from \"./components/GifGrid\";\r\n\r\nexport const GifExpertApp = () => {\r\n\t// const categories = [\"One Punch\", \"Samurai X\", \"Dragon ball\"];\r\n\tconst [categories, setCategories] = useState(['Busca un gif']);\r\n\r\n\t// const handleAdd = () =>{\r\n\t//     const element = 'Jei';\r\n\t//     // setCategories([...categories,element]); //Una solucion\r\n\t//     setCategories( c => [...c, element]);\r\n\t// };\r\n\r\n\treturn (\r\n\t\t<>\r\n\t\t\t<h2>GifExpertApp</h2>\r\n\t\t\t<AddCategory categories={categories} setCategories={setCategories} />\r\n\t\t\t<hr />\r\n\t\t\t{/* <button onClick={handleAdd}>Agregar</button> */}\r\n\t\t\t<ol>\r\n\t\t\t\t{categories.map((category) => (\r\n\t\t\t\t\t<GifGrid key={category} category={category} />\r\n\t\t\t\t))}\r\n\t\t\t</ol>\r\n\t\t</>\r\n\t);\r\n};\r\n","import React from \"react\";\nimport ReactDOM from \"react-dom\";\nimport {GifExpertApp} from \"./GifExpertApp\";\n\nimport \"./index.css\";\n\nReactDOM.render(<GifExpertApp />, document.getElementById(\"root\"));\n"],"sourceRoot":""}