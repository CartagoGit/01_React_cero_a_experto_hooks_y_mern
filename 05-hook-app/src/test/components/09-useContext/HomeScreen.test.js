/* eslint-disable */

import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen.js";
import { UserContext } from "../../../components/09-useContext/UserContext.js";

describe("Pruebas en <HomeScreen />>", () => {
    const user = {
        name: 'Mario',
        email: 'unemail@cualquiera.com'
    }
	let wrapper;
	const fillWrapper = () => {
		wrapper = mount(
			<UserContext.Provider value={{user}}>
				<HomeScreen />
			</UserContext.Provider>
		);
	};

	beforeEach(() => fillWrapper());

	test("Deberia mostrarse correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});
});
