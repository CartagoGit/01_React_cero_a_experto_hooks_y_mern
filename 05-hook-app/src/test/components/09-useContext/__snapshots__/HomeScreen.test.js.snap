// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen />> Deberia mostrarse correctamente 1`] = `
Array [
  <h1>
    Home Screen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "Mario",
   "email": "unemail@cualquiera.com"
}
  </pre>,
]
`;
